lib/Driver/Driver.cpp

```diff
@@ -663,6 +660,11 @@ Driver::buildCompilation(const ToolChain &TC,
         rebuildEverything = true;
 
       } else {
+        // In 'emit-module' only mode, use '~moduleonly' suffixed build record
+        // file.
+        if (OI.CompilerOutputType == file_types::TY_SwiftModuleFile)
+          buildRecordPath = buildRecordPath.append("~moduleonly");
+
```

Comment:
I had to look at the entire file to absorb this change, and then had to look at the whole code block starting with `InputInfoMap outOfDateMap;`. I think that these very long methods such as `buildCompilation` add to the effort required for a noobie (such as me) to understand the code. At some point as these methods grow, it behooves us to break them up, to help comprehensibility. Since you are modifying this code, you have a chance to improve this part of the situation, rather than making the method incrementally longer. Would you consider pulling this code block out into a subroutine? Either a static function or a member function? As far as I can tell, its purpose is to populate the out of date map, and return an indication of whether everything should be rebuilt. Thanks!